src/main/java/graphql/GraphQL.java

```diff
@@ -27,6 +28,18 @@
     private final GraphQLSchema graphQLSchema;
     private final ExecutionStrategy queryStrategy;
     private final ExecutionStrategy mutationStrategy;
+    //
+    // later PR changes will allow api consumers to provide their own id provider
+    //
+    // see https://github.com/graphql-java/graphql-java/pull/276 for the builder pattern
+    // needed to make this sustainable.  But for now we will use a hard coded approach.
+    //
+    private final ExecutionIdProvider idProvider = new ExecutionIdProvider() {
+        @Override
+        public ExecutionId generate(String query, String operationName, Object context) {
```

Comment:
How about calling this `provide()` since this may providers may provide an id not by generating it?